execution/engine/config_factory_federation.go

```diff
@@ -150,6 +157,44 @@ func (f *FederationEngineConfigFactory) BuildEngineConfiguration() (conf Configu
 	return conf, nil
 }
 
+// Compose produces a federated router configuration.
+func (f *FederationEngineConfigFactory) Compose() (*RouterConfig, error) {
+	res, err := f.compose()
+	if err != nil {
+		return nil, err
+	}
+	return &RouterConfig{routerConfig: res}, nil
+}
+
+// RouterConfig is an intermediate data that holds the result of the composition
+// and can be saved to a file to configure the router data sources when creating an engine config.
+type RouterConfig struct {
```

Comment:
Hi @ikawaha 

We probably could use proto types directly
Are there any special need in the wrapping structure?

Reply:
Thank you for your feedback. I initially wrapped nodev1.RouterConfig because it was hidden, and my goal was to make it serializable. However, I agree that using the proto types directly would be clearer and more straightforward. I've made the necessary adjustments to avoid the wrapping structure and now utilize nodev1.RouterConfig directly.

I've pushed the changes—does this look better to you? [24b4a97](https://github.com/wundergraph/graphql-go-tools/pull/876/commits/24b4a97049e34885563ccfa9977fa727cffefa64)